File: pueblo-golf-tournament-api/Entities/Person.cs
using pueblo_golf_tournament_api.Base;

namespace pueblo_golf_tournament_api.Entities
{
    public class Person : BaseEntity
    {
        public string? FirstName { get; set; }
        public string? MiddleName { get; set; }
        public string? LastName { get; set; }
        public DateTime BirthDate { get; set; }
        public string? ContactNumber { get; set; }
        public string? EmailAddress { get; set; }
        public string? Country { get; set; }
    }
}

File: pueblo_golf_tournament_mobile/lib/models/person.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pueblo_golf_tournament_mobile/models/base-model.dart';
import 'package:pueblo_golf_tournament_mobile/utilities/enums.dart';
part 'person.g.dart';

@JsonSerializable()
class Person extends BaseModel {
  final String firstName;
  final String middleName;
  final String lastName;
  final DateTime birthDate;
  final String contactNumber;
  final String emailAddress;
  final String country;

  factory Person.fromJson(Map<String, dynamic> json) => _$PersonFromJson(json);

  Person(
      {required this.firstName,
      required this.middleName,
      required this.lastName,
      required this.birthDate,
      required this.contactNumber,
      required this.emailAddress,
      required this.country,
      required super.id});
  Map<String, dynamic> toJson() => _$PersonToJson(this);
}
